### PR TITLE
Fixed #33662 -- Allowed Sitemap to customize languages for each item.

### PR DESCRIPTION
**Pull request by roxanebellot**
[ticket #33662 :  Choose which items are displayed per language in Sitemap ](https://code.djangoproject.com/ticket/33662)

### CODE REVIEW

File: django/contrib/sitemaps/__init__.py [hunk 1]
```diff
@@ -202,23 +209,25 @@ def _urls(self, page, protocol, domain):
...
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    loc = loc.replace(f"/{lang_code}/", "/", 1)
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": "x-default",
-                        }
-                    )
+                    if self.enable_item_for_lang(item[0], lang_code):
+                        lang_code = settings.LANGUAGE_CODE
```
Comment (felixxm) [inline django/contrib/sitemaps/__init__.py:222]: `lang_code` should be set before the check:
```suggestion
                    lang_code = settings.LANGUAGE_CODE
                    if self.enable_item_for_lang(item[0], lang_code):
```

File: django/contrib/sitemaps/__init__.py [hunk 2]
```diff
@@ -202,23 +209,25 @@ def _urls(self, page, protocol, domain):
...
             if self.i18n and self.alternates:
                 for lang_code in self._languages():
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": lang_code,
-                        }
-                    )
+                    if self.enable_item_for_lang(item[0], lang_code):
```
Comment (felixxm) [inline django/contrib/sitemaps/__init__.py:212]: What do you think about adding a method with languages for the item instead? E.g.
```python
def get_languages_for_item(self, item):
    return self._languages()

def _urls(self, page, protocol, domain):
    ...
    if self.i18n and self.alternates:
        for lang_code in self.get_languages_for_item(item):
```

File: docs/releases/4.2.txt [hunk 3]
```diff
@@ -121,7 +121,9 @@ Minor features
 :mod:`django.contrib.sitemaps`
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* ...
+* :class:`~django.contrib.sitemaps.Sitemap` can now choose witch items
```
Comment (felixxm) [inline docs/releases/4.2.txt:124]: :broom: _"witch"_

**Review by felixxm — COMMENTED** [review]
@roxanebellot Thanks :+1: Please check my comments.

File: django/contrib/sitemaps/__init__.py (hunk 2 above)
Comment (roxanebellot) [reply django/contrib/sitemaps/__init__.py]: Love it !

File: docs/releases/4.2.txt (hunk 3 above)
Comment (roxanebellot) [reply docs/releases/4.2.txt]: Haha can't even blame it on the season. Sorry, well seen !

File: django/contrib/sitemaps/__init__.py [hunk 4]
```diff
@@ -202,23 +208,25 @@ def _urls(self, page, protocol, domain):
...
             if self.i18n and self.alternates:
                 for lang_code in self._languages():
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": lang_code,
-                        }
-                    )
+                    if lang_code in self.get_languages_for_item(item[0]):
```
Comment (felixxm) [inline django/contrib/sitemaps/__init__.py:211]: Why don't do this in a single line? I think we can also call `get_languages_for_item()` only once:
```python
if self.i18n and self.alternates:
    languages = self.get_languages_for_item(item[0])
    for lang_code in languages:
        loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
        ...

    if self.x_default and settings.LANGUAGE_CODE in languages:
        lang_code = settings.LANGUAGE_CODE
        loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
        ...
```

Comment (roxanebellot) [reply django/contrib/sitemaps/__init__.py:211]: I followed your advice. If you have time, could you explain how 
```python
if self.x_default and settings.LANGUAGE_CODE in languages:
    lang_code = settings.LANGUAGE_CODE
```
is better than
```python
lang_code = settings.LANGUAGE_CODE
if self.x_default and lang_code in languages:
```

please ?

Comment (felixxm) [reply django/contrib/sitemaps/__init__.py]: One assignment less, not much difference, TBH.

File: django/contrib/sitemaps/__init__.py [hunk 5]
```diff
@@ -201,15 +207,18 @@ def _urls(self, page, protocol, domain):
...
+                    {
+                        "location": (
+                            f"{protocol}://{domain}{self._location(item, lang_code)}"
+                        ),
+                        "lang_code": lang_code,
+                    }
+                    for lang_code in item_languages
+                ]
+
+                if self.x_default and settings.LANGUAGE_CODE in item_languages:
```
Comment (felixxm) [inline django/contrib/sitemaps/__init__.py:221]: I added test for this change.

**Review by felixxm — COMMENTED** [review]
@roxanebellot Thanks :+1: 

I pushed final edits.

**Pull request merged**